chaosgcp/iam/controls/policy.py

```diff
@@ -0,0 +1,304 @@
+            roles=["roles/editor"],
+            roles_iam_condition=condition,
+            manage_type="add",
+        )
+    """
+    # Type checks for arguments
+    if not isinstance(project_id, str):
+        raise TypeError("project_id must be a string")
+    if not isinstance(members, list):
+        raise TypeError("members must be a list of strings")
+    if not isinstance(roles, list):
+        raise TypeError("roles must be a list of strings")
+    if not isinstance(roles_iam_condition, dict):
+        raise TypeError("roles_iam_condition must be a dictionary")
+    if not isinstance(manage_type, str):
+        raise TypeError("manage_type must be a string")
+    if not isinstance(iam_propogation_sleep_time_in_minutes, int):
+        raise TypeError(
+            "iam_propogation_sleep_time_in_minutes must be an integer"
+        )
+
+    # Initializes service.
+    crm_service = initialize_service()
+
+    if manage_type == "add":
+        logger.info("Adding Conditional Roles..")
+        logger.info(roles_iam_condition)
+        modify_policy_add_roles(
+            crm_service,
+            project_id,
+            roles,
+            members,
+            roles_iam_condition,
+            iam_propogation_sleep_time_in_minutes,
+        )
+
+    elif manage_type == "remove":
+        logger.info(roles_iam_condition)
+        logger.info("Removing Conditional Roles..")
+        modify_policy_remove_roles(
+            crm_service, project_id, roles, members, roles_iam_condition
+        )
+
+    else:
+        logger.warning("Invalid manage type: must be 'add' or 'remove'.")
+        raise ValueError("Invalid manage type: must be 'add' or 'remove'")
+
+
+def initialize_service() -> dict:
+    """Initializes a Cloud Resource Manager service."""
+
+    credentials, _ = google.auth.default(
+        scopes=["https://www.googleapis.com/auth/cloud-platform"]
+    )
+    crm_service = googleapiclient.discovery.build(
+        "cloudresourcemanager", "v1", credentials=credentials
+    )
+    return crm_service
+
+
+def get_policy(crm_service, project_id: str, version: int = 3) -> dict:
+    """Gets IAM policy for a project."""
+    logger.info("Getting IAM policy for a project.")
+    policy = (
+        crm_service.projects()
+        .getIamPolicy(
+            resource=project_id,
+            body={"options": {"requestedPolicyVersion": version}},
+        )
+        .execute()
+    )
+    return policy
+
+
+def set_policy(crm_service, project_id: str, policy: str) -> dict:
+    """Sets IAM policy for a project."""
+    logger.info("Setting IAM policy for a project.")
+    policy = (
+        crm_service.projects()
+        .setIamPolicy(resource=project_id, body={"policy": policy})
+        .execute()
+    )
+    return policy
+
+
+def modify_policy_add_roles(
+    crm_service,
+    project_id: str,
+    roles: list[str],
+    member: list[str],
+    iam_condition: dict[str, str],
+    iam_propogation_sleep_time_in_minutes: int,
```

Comment:
fix the typing annotation to be supported by Python 3.8+

Reply:
changed to List and Dict .